napari/utils/progress.py

```diff
@@ -0,0 +1,245 @@
+import inspect
+from contextvars import ContextVar
+from typing import Iterable, Optional
+
+from qtpy import QtCore
+from qtpy.QtWidgets import (
+    QApplication,
+    QHBoxLayout,
+    QLabel,
+    QProgressBar,
+    QVBoxLayout,
+    QWidget,
+)
+from tqdm import tqdm
+
+from .._qt.utils import get_viewer_instance
+
+IS_NESTED = ContextVar('IS_NESTED', default=False)
+
+
+def get_pbar(viewer_instance, **kwargs):
+    """Adds ProgressBar to viewer Activity Dock and returns it.
+
+    Parameters
+    ----------
+    viewer_instance : qtViewer
+        current napari qtViewer instance
+
+    Returns
+    -------
+    ProgressBar
+        progress bar to associate with current iterable
+    """
+    pbar = ProgressBar(**kwargs)
+    pbr_layout = viewer_instance.activityDock.widget().layout
+    if IS_NESTED.get():
```

Comment:
rather than making `IS_NESTED` boolean and then asking the layout for the previous progressbar, let's have `IS_NESTED` store that information directly.

Reply:
@tlambert03 this definitely seems like it would be an improvement but how do we then dereference the `id(self)` or even `id(self._pbar)` into the widget we need? Wouldn't we still have to hunt for it through the possible `ProgressBarGroup`s?

Reply:
you still might need to hunt... but at least with an id you know _what_ you're hunting for (as opposed to a boolean where you have to just assume it's the last one).  You can also use a weakref if you don't want to hunt